cdc/model/changefeed.go

```diff
@@ -217,19 +219,45 @@ func (info *ChangeFeedInfo) CheckErrorHistory() (needSave bool, canInit bool) {
 		ts := info.ErrorHis[i]
 		return time.Since(time.Unix(ts/1e3, (ts%1e3)*1e6)) < ErrorHistoryGCInterval
 	})
-	if i == len(info.ErrorHis) {
-		info.ErrorHis = info.ErrorHis[:]
-	} else {
-		info.ErrorHis = info.ErrorHis[i:]
-	}
+	info.ErrorHis = info.ErrorHis[i:]
+
 	if i > 0 {
 		needSave = true
 	}
 
 	i = sort.Search(len(info.ErrorHis), func(i int) bool {
 		ts := info.ErrorHis[i]
-		return time.Since(time.Unix(ts/1e3, (ts%1e3)*1e6)) < errorHistoryCheckInterval
+		return time.Since(time.Unix(ts/1e3, (ts%1e3)*1e6)) < ErrorHistoryCheckInterval
 	})
-	canInit = len(info.ErrorHis)-i < errorHistoryThreshold
+	canInit = len(info.ErrorHis)-i < ErrorHistoryThreshold
 	return
 }
+
+// HasFastFailError returns true if the error in changefeed is fast-fail
+func (info *ChangeFeedInfo) HasFastFailError() bool {
+	if info.Error == nil {
+		return false
+	}
+	return filter.ChangefeedFastFailErrorCode(errors.RFCErrorCode(info.Error.Code))
+}
+
+// CheckErrorHistoryV2 checks error history of a changefeed
+// if having error record older than GC interval, set needSave to true.
+// if error counts reach threshold, set canInit to false.
+func (info *ChangeFeedInfo) CheckErrorHistoryV2() (canInit bool) {
```

Comment:
```suggestion
func (info *ChangeFeedInfo) CheckErrorHistoryV2() bool {
```

Seems the named return parameter was useless.